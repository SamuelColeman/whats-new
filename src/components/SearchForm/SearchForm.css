input {
	width: 80%;
  margin: 1% 0;
  border: none;
  background: #141527;
  height: 6vh;
  font-size: 20px;
  color: white;
}

input:focus {
	outline: 0;
}

form {
	width: 100vh;
}

.searchBtn {
	background: #B56AF0;
	height: 4vh;
  width: 16vh;
  border-radius: 15px;
  font-size: 15px;
}

.searchBtn:hover {
	background: #FFAD03;
	box-shadow: 5px 5px 5px black;
	border-radius: 15px;
}