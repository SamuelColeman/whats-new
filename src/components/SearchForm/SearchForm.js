import React, { Component } from 'react';
import './SearchForm.css';

class SearchForm extends Component {
	constructor(props) {
		super(props);
		this.state = { search: ''}
	}

	handleChange = event => {
		event.preventDefault();
		this.props.searchArticles(this.state.search);
	}

	updateSearch = event => {
		this.setState({ [event.target.name]: event.target.value})
	}

	render() {
		return (
			<form>
				<input onChange={this.updateSearch} 
					type="text"
					name="search"
					value={this.state.search}
					placeholder="Article"/>
				<button type="submit" onClick={this.handleChange} onSubmit={this.handleChange}>Search
				</button>
			</form>
		)
	}
}

export default SearchForm;