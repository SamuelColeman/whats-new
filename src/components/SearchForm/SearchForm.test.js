import React from 'react';
import { shallow } from 'enzyme'
import SearchForm from './SearchForm';

describe('SearchForm', () => {
	const mockSearchArticles = jest.fn();
	const mockEvent = {target: {name: 'search', value: 'infection'}};
	const wrapper = shallow(<SearchForm searchArticles={mockSearchArticles} />)

	it('should match the snapshot', () => {
		expect(wrapper).toMatchSnapshot();
	});

	it('should invoke searchArticles when handleChange is called', () => {
		const mockEvent = { preventDefault: jest.fn() };
		wrapper.instance().handleChange(mockEvent);
		expect(mockSearchArticles).toHaveBeenCalledWith(wrapper.state('search'))
	});

	it ('should update state when updateSearch is invoked', () => {
		wrapper.instance().updateSearch(mockEvent);
		expect(wrapper.state('search')).toEqual(mockEvent.target.value);
	});

	it('should invoke updateSearch when the input value is changed', () => {
		wrapper.instance().updateSearch = jest.fn();
		wrapper.instance().forceUpdate();
		wrapper.find('input').simulate('change', mockEvent);
		expect(wrapper.instance().updateSearch).toHaveBeenCalledWith(mockEvent);
	})

	it('should invoke handleChange when the button is clicked', () => {
		wrapper.instance().handleChange = jest.fn();
		wrapper.instance().forceUpdate();
		wrapper.find('button').simulate('click', mockEvent);
		expect(wrapper.instance().handleChange).toHaveBeenCalledWith(mockEvent);
	})
})  