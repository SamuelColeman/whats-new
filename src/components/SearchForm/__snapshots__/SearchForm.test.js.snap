// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchForm should match the snapshot 1`] = `
<form>
  <input
    name="search"
    onChange={[Function]}
    placeholder="Article"
    type="text"
    value=""
  />
  <button
    onClick={[Function]}
  >
    Search
  </button>
</form>
`;
